crio.conf(5): update manpage to the latest state

vrothberg opened the pull request:
Update the crio.conf(5) manpage to reflect the latest state of CLI flags
and configuration options.  Also reflect those changes in crio-config.

Fixes: #1747
Signed-off-by: Valentin Rothberg <vrothberg@suse.com>

TomSweeneyRedHat commented:
/ok-to-test

vrothberg commented:
@mrunalp @rhatdan PTAL

I found some inconsistencies and format issues in `crio config` that I can fix once I have your ACK on the proposed changes to the manpage. After that, I will update `crio config` to be as consistent with the manpage as possible.

rhatdan commented on docs/crio.conf.5.md:
```diff
@@ -1,19 +1,17 @@
-% crio.conf(5) Open Container Initiative Daemon
+% crio.conf(5) Open Container Initiative Kubernetes Container Runtime daemon
```
Lets call this
Kubernetes Container Runtime Daemon for Open Container Initiative Containers

TomSweeneyRedHat replied:
Other man pages keep this short and sweet.  I'll let you all thumb wrestle over it, but I'd suggest just simply:  
 
Configuration file for CRI-O

Then go with the full name at line 6.  FWIW, in GitHub we call it "Open Container Initiative-based implementation of Kubernetes Container Runtime Interface".  I'd suggest line 6 change to:  

"configuration file for the Open Container Initiative-based implementation of Kubernetes Container Runtime Interface"

rhatdan replied:
I would rather get the Kubernetes and Container into this line so that
man -k kubernetes
man -k container

Would list cri-o.conf

rhatdan commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
-  If it is positive, it must be >= 8192 (to match/exceed conmon read buffer).
-  The file is truncated and re-opened so the limit is never exceeded.
+**conmon**="/usr/local/libexec/crio/conmon"
+  Path to the conmon binary, used for managing the runtime.
```
user for monitoring/managing the OCI Runtime.

rhatdan commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
 
-**pause_image**=""
-  Image which contains the pause executable (default: "kubernetes/pause")
+**registries**=["docker.io"]
```
Can you add some other registries to this list.  Even if they are fake.

vrothberg replied:
All assignments indicate the default values, so I am rather hesitant to add more. Are you also okay with giving an example in the description mentioning some other registries?

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
     option = value
 
 ## CRIO TABLE
+CRI-O reads its storage defaults from the containers-storage.conf(5) file located at /etc/containers/storage.conf. Modify this storage configuration if you want to change the system's defaults. If you want to modify just CRI-O, you can change the storage configuration options here.
```
suggest: "modify just CRI-O" to "modify storage just for CRI-O"

rhatdan replied:
I agree.

vrothberg replied:
I opened https://github.com/containers/storage/issues/209 for that and optimistically made the that statement already here :-D

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
-**runroot**=""
-  CRIO state dir (default: "/var/run/containers/storage")
+**storage_option**=[]
+  List to pass options to the storage driver. Please refer to containers-storage.conf(5) to see all available storage options.
```
I'm not seeing "all available" options/drivers in my man page for containers-storage.conf.  Is it out of date and/or will be updated?

rhatdan replied:
We should update it.

vrothberg replied:
I opened https://github.com/containers/storage/issues/209 for that and optimistically made the that statement already here :-D

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
 
-## CRIO.API TABLE
+**runtime**="/usr/bin/runc"
+  Path to the OCI compatible runtime used for trusted container workloads. This is a mandatory setting as this runtime will be the default one and will also be used for untrusted container workloads if `runtime_untrusted_workload` is not set.
```
suggest "default one and" to "default and"

rhatdan replied:
:+1:

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
+**default_capabilities**=[]
+  List of default capabilities for containers. If unset (e.g., commented out), the default list of capabilities defined in the spec will be added. If `capabilities` is empty (i.e., set to `[]`), only the capabilities defined in the container json file by the user/kube will be added.
+
+  The default list looks as follows:
```
suggest "The default list is:"

rhatdan replied:
:+1:

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
 
-**no_pivot**=*true*|*false*
-  Instructs the runtime to not use pivot_root, but instead use MS_MOVE
+    2) `/usr/share/containers/mounts.conf`: This is the default file read for mounts. If you want CRI-O to read from a different, specific mounts file, you can change the default_mounts_file path right below. Note, if this is done, CRI-O will only add mounts it finds in this file.
```
The default_mounts_file path is not "right below" in this doc.  I'd remove "right below".

rhatdan replied:
:+1:

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
+  Maximum number of processes allowed in a container.
+
+**log_size_max**=-1
+  Maximum sized allowed for the container log file. Negative numbers indicate that no size limit is imposed. If it is positive, it must be >= 8192 to match/exceed conmon's read buffer. The file is truncated and re-opened so the limit is never exceeded.
```
I'll take that back, we briefly discuss conmon up near the top.  It still would be nice to have a man page for that, and lacking that, perhaps a bit more explanation of what conmon is above.

rhatdan replied:
We have a conmon package we should have a conmon man page.

vrothberg replied:
I suggest to wait until `conmon(1)` is there and update it here in the same PR.

TomSweeneyRedHat replied:
I've created an issue on CRI-O to create a conmon page and to then circle back and touch up this page once created.  I'm fine with just leaving confmon references in this page until that happens.

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
-**default_transport**
-  A prefix to prepend to image names that can't be pulled as-is (default: "docker://")
+**default_transport**="docker://"
+  A prefix we for images that cannot be parsed as a valid source reference.
```
not sure what you're trying to say in this sentence.  Perhaps drop "we"?

rhatdan replied:
`Default transport for pulling images from remote container storage.`

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
-  Enable insecure registry  communication,  i.e.,  enable  un-encrypted
-  and/or untrusted communication.
+**pause_command**="/pause"
+  The command to run in a `pause_image` to have a container just sit there. If the image contains the necessary information, this value does not need to be specified.
```
Suggest "The command to run to have a container stay in the paused state."

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
-  to specify a whole  subnet.  Insecure  registries  accept  HTTP  and/or
-  accept HTTPS with certificates from unknown CAs.
+**signature_policy**="/etc/containers/policy.json"
+  Path the file which decides what sort of policy we use when deciding whether or not to trust an image that we've pulled. Please refer to containers-policy.json(5) for more details.
```
in our other docs we've added a warning to not change this.  For instance Buildah/Podman do:

Pathname  of  a  signature  policy  file to use.  It is not recommended that this option be used, as the
       
Pathname  of  a  signature  policy  file to use.  It is not recommended that this option be used, as the
default behavior of using the system-wide default  policy  (frequently  /etc/containers/policy.json)  is
most often preferred.

It would probably be good to layer such a warning here.

TomSweeneyRedHat commented on docs/crio.conf.5.md:
```diff
@@ -25,143 +23,188 @@ No bare options are used. The format of TOML can be simplified to:
+**registries**=["docker.io"]
+  List of registries to be used when pulling an unqualified image (e.g., "alpine:latest").
+
+CRI-O reads its configured registries defaults from the system wide containers-registries.conf(5) located in /etc/containers/registries.conf. If you want to modify just CRI-O, you can change the registies configuration in this file. Otherwise, leave `insecure_registries` and `registries` comment out to use system's defaults.
```
suggest: "comment out to use system's defaults." to "commented out to use the system's defaults from /etc/containers/registries.conf."

vrothberg commented:
@TomSweeneyRedHat, thank you so much for such a detailed review, much appreciated! I will address all your comments tomorrow (ECST tomorrow) with a fresh pair of eyes.

vrothberg commented:
@mrunalp @rhatdan @TomSweeneyRedHat, thanks a lot for the great reviews. I addressed your comments and updated the commit.

I want to collect your opinion of what to do with the `crio-config` command. I extracted most of the text from there and there's obviously some work to improve it as well. But before doing that, I want to throw out the idea removing `crio-config` and rather focus on the manpage in the future. If you want to keep `crio-config`, I'll happily reflect the changes there.

vrothberg commented:
I went ahead and changed the template for `crio-config` as well so that the manpage and output of `crio-config` and in sync. I also dropped two reminder to the source in the hope that the manpage and the template don't diverge to much in the future.

@TomSweeneyRedHat @mrunalp @rhatdan PTAL

vrothberg commented:
@runcom PTAL :)

TomSweeneyRedHat commented on cmd/crio/config.go:
```diff
@@ -8,249 +8,269 @@ import (
+
+# CRI-O reads its storage defaults from the containers-storage.conf(5) file
+# located at /etc/containers/storage.conf. Modify this storage configuration if
+# you want to change the system's defaults. If you want to modify storagae just
```
"storagae" to "storage"

TomSweeneyRedHat commented:
One small nit and would like @runcom and/or @mrunalp to take a look too.  Travis isn't happy at the moment fwiw.  Otherwise LGTM.

vrothberg commented:
Thanks, @TomSweeneyRedHat. I updated the manpage and the template in crio-config. Travis fails with a vet `wrapf` format error somewhere else; it's fixed in another PR.

mrunalp commented:
Needs rebase, then we can merge.

runcom commented:
/test all

vrothberg commented:
It actually requires a merge now :(

vrothberg commented:
@mrunalp I resolved the merged conflict and synced crio-config and the manpage. Let's get this off the table.

mrunalp commented:
@vrothberg Yeah, will get it in as soon as travis passes.

mrunalp commented:
/test all
